src/mpid/ch4/src/ch4r_callbacks.c

```diff
@@ -38,6 +38,41 @@ int MPIDIG_do_long_ack(MPIR_Request * rreq)
     goto fn_exit;
 }
 
+int MPIDIG_do_cts(MPIR_Request * rreq)
+{
+    int mpi_errno = MPI_SUCCESS;
+
+    MPL_DBG_MSG_FMT(MPIDI_CH4_DBG_GENERAL, VERBOSE,
+                    (MPL_DBG_FDEST, "do cts req %p handle=0x%x", rreq, rreq->handle));
+
+    MPIDIG_data_cts_msg_t am_hdr;
+    am_hdr.sreq_ptr = (MPIDIG_REQUEST(rreq, req->rreq.peer_req_ptr));
+    am_hdr.rreq_ptr = rreq;
+    MPIR_Assert((void *) am_hdr.sreq_ptr != NULL);
+
+#ifdef MPIDI_CH4_DIRECT_NETMOD
+    mpi_errno = MPIDI_NM_am_send_hdr_reply(MPIDIG_REQUEST(rreq, context_id),
+                                           MPIDIG_REQUEST(rreq, rank), MPIDIG_DATA_CTS, &am_hdr,
+                                           sizeof(am_hdr));
+#else
+    if (MPIDI_REQUEST(rreq, is_local)) {
+        mpi_errno = MPIDI_SHM_am_send_hdr_reply(MPIDIG_REQUEST(rreq, context_id),
+                                                MPIDIG_REQUEST(rreq, rank),
+                                                MPIDIG_DATA_CTS, &am_hdr, sizeof(am_hdr));
+    } else {
+        mpi_errno = MPIDI_NM_am_send_hdr_reply(MPIDIG_REQUEST(rreq, context_id),
+                                               MPIDIG_REQUEST(rreq, rank),
+                                               MPIDIG_DATA_CTS, &am_hdr, sizeof(am_hdr));
+    }
+#endif
+    MPIR_ERR_CHECK(mpi_errno);
+
+  fn_exit:
+    return mpi_errno;
+  fn_fail:
+    goto fn_exit;
+}
+
```

Comment:
Now that `RTS` is the old rendezvous `MPIDIG_SEND_LONG_REQ`, this CTS is the same as the old `MPIDIG_SEND_LONG_ACK`, right?

Reply:
`MPIDIG_SEND_LONG_REQ` is not needed. I would say we remove it as part of the PR. As as `MPIDIG_SEND_LONG_ACK`.

Reply:
We don't need `MPIDIG_SEND_LONG_REQ` in the same way we don't need `MPIDIG_SSEND_REQ`. It is just `MPIDIG_SEND`